alvr/server_openvr/src/lib.rs

```diff
@@ -123,117 +120,121 @@ extern "C" fn driver_ready_idle(set_default_chap: bool) {
+
+                            (
+                                hand_skeleton_config.steamvr_input_2_0,
+                                tracked.then_some(left_hand_skeleton).flatten(),
+                                tracked.then_some(right_hand_skeleton).flatten(),
+                            )
                         } else {
-                            ptr::null()
-                        },
-                        handSkeleton: if let Some(skeleton) = &ffi_left_hand_skeleton {
-                            skeleton
-                        } else {
-                            ptr::null()
-                        },
-                        isHandTracker: use_separate_hand_trackers
-                            && ffi_left_controller_motion.is_none()
-                            && ffi_left_hand_skeleton.is_some(),
-                    };
-                    let right_hand_data = FfiHandData {
-                        controllerMotion: if let Some(motion) = &ffi_right_controller_motion {
-                            motion
-                        } else {
-                            ptr::null()
-                        },
-                        handSkeleton: if let Some(skeleton) = &ffi_right_hand_skeleton {
-                            skeleton
-                        } else {
-                            ptr::null()
-                        },
-                        isHandTracker: use_separate_hand_trackers
-                            && ffi_right_controller_motion.is_none()
-                            && ffi_right_hand_skeleton.is_some(),
-                    };
-
-                    let ffi_body_trackers =
-                        tracking::to_ffi_body_trackers(&tracking.device_motions, track_body);
+                            (false, None, None)
+                        };
 
-                    // There are two pairs of controllers/hand tracking devices registered in
-                    // OpenVR, two lefts and two rights. If enabled with use_separate_hand_trackers,
-                    // we select at runtime which device to use (selected for left and right hand
-                    // independently. Selection is done by setting deviceIsConnected.
-                    unsafe {
-                        SetTracking(
-                            tracking.target_timestamp.as_nanos() as _,
-                            controllers_pose_time_offset.as_secs_f32(),
-                            ffi_head_motion,
-                            left_hand_data,
-                            right_hand_data,
-                            if let Some(body_trackers) = &ffi_body_trackers {
-                                body_trackers.as_ptr()
+                        let ffi_left_hand_data = FfiHandData {
+                            controllerMotion: if let Some(motion) = &ffi_left_controller_motion {
+                                motion
                             } else {
                                 ptr::null()
```

Comment:
sure, i'll do

Reply:
I'm trying now and i struggled to use the functional pattern. The problem is the first branch has a implicit cast from reference to raw pointer. also I need to make sure the original object stays alive otherwise it would be a dangling pointer (well, the compiler just won't make me cast a local reference to pointer because it's UB). so the solution I came up with is:
```
controllerMotion: ffi_left_controller_motion
      .as_ref()
      .map(|m| m as *const _)
      .unwrap_or(ptr::null()),
```
 which to be honest seems even more verbose.